### DISCUSSION TITLE
v4.2.0

### DISCUSSION THREAD

**Discussion by Harmon758** [tweepy/tweepy#1694] [Announcements]
## New Features / Improvements
- Add support for managing lists with Twitter API v2 ([b1342bf](https://github.com/tweepy/tweepy/commit/b1342bfc998bee334437f2b7a8d2aef4df7c3838))
- Rename `Client.follow` and `Client.unfollow` to `Client.follow_user` and `Client.unfollow_user`, respectively ([8f8de15](https://github.com/tweepy/tweepy/commit/8f8de15e13f11d042a521d2adae24d0c09fa2098))
  - `Client.follow` and `Client.unfollow` are kept as deprecated aliases
- Add FAQ section in documentation
- Update and improve various documentation and tests

### Twitter API Changes
- Change `state` to optional parameter for `Client.search_spaces` ([e61d5d6](https://github.com/tweepy/tweepy/commit/e61d5d6f017db7f28871331b515fcfd87666f352))

## Bug Fixes
- Fix parsing of datetime strings for API v2 models with Python 3.6 ([5bf2446](https://github.com/tweepy/tweepy/commit/5bf24464b00257a9fa5f66047a2f7815c1e4f8fb))
- Fix models missing an `API` instance attribute when using `Cursor` with pagination by ID ([451e921](https://github.com/tweepy/tweepy/commit/451e921210677ee0a618849f189bdfeea497a00c))

<hr /><em>This discussion was created from the release <a href='https://github.com/tweepy/tweepy/releases/tag/v4.2.0'>4.2.0</a>.</em>

**Comment by jdc4429** [comment]
Could we please get an example of the Stream function in python.  Everything I'm finding on the web is old and no longer works.

**Reply by Harmon758** [reply]
This is pretty high priority for me right now, especially with https://github.com/tweepy/tweepy/commit/5fa6e73bbe677cf2527af7187012f665b825639d done.
I'll have an example for it committed soon.

**Reply by Harmon758** [reply]
For now though, see [the reference documentation for `Stream`](https://tweepy.readthedocs.io/en/v4.2.0/stream.html) and [the page on streaming in the documentation](https://tweepy.readthedocs.io/en/v4.2.0/streaming.html).

**Reply by Harmon758** [reply]
I've added a basic example of streaming, with https://github.com/tweepy/tweepy/commit/56f15473917155a44c29e49c97cba1ef4cef159d, based on the existing documentation.

**Comment by jdc4429** [comment]
Thank you for your help and effort in this project, I can work with that
example.  I appreciate the quick reply.
If you could indulge me with one more question, is it possible to stream
from a list? Ultimately I'm trying to create a list of influencers and
stream new tweets from the people on the list to examine sentiment over a
given time frame.

Thanks again,
   Jeff

**Reply by Harmon758** [reply]
Yes, [`Stream.filter`](https://tweepy.readthedocs.io/en/v4.2.0/stream.html#tweepy.Stream.filter) accepts a list of user IDs for the `follow` parameter.

**Comment by jdc4429** [comment]
Thank you again, you're awesome.

I am still learning Python so it's difficult to do anything without
examples. I'm not sure how to reference the list itself...

follow (*Optional**[**List*
<https://docs.tweepy.org/en/v4.2.0/models.html#tweepy.models.List>*[**Union*
*[**int* <https://docs.python.org/3/library/functions.html#int>*, **str*
<https://docs.python.org/3/library/stdtypes.html#str>*]**]**]*)

Do I just put the list ID? I have no idea what Union does and it seems this
doesn't work.

streamit.filter(follow=["1454607810474426375"], languages=["en"])

**Reply by Harmon758** [reply]
The `Union` just means you can specify the IDs as either integers or strings.

This should work if `streamit` is defined properly and `1454607810474426375` is a valid user ID.

Did you mean a Twitter list or a Python list?
If you meant a Twitter list, you'll need to retrieve the members of that list first.
See [the documentation for `API.get_list_members`](https://tweepy.readthedocs.io/en/v4.2.0/api.html#tweepy.API.get_list_members).

**Comment by jdc4429** [comment]
Thank you, I just figured that out.  I was assuming that you could just
enter a list_id to follow.  Didn't realize that you needed to pull all the
id's in the list and then send those.  What happens when you assume. lol

Thanks again for all your help.  Enjoy your day.
